Tests/MaterialLib/TestRelativePermeabilityModel.cpp

```diff
@@ -0,0 +1,201 @@
+{
+    const char xml[] =
+        "<relative_permeability id=\"0\">"
+        "   <type>WettingPhaseBrookCoreyOilGas</type>"
+        "   <sr>  0.2 </sr> "
+        "   <smax> 0.8 </smax> "
+        "   <m> 2 </m> "
+        "   <krel_min> 1.e-9 </krel_min> "
+        "</relative_permeability>";
+    auto const perm_model = createRelativePermeabilityModel(xml);
+
+    std::vector<double> S = {0.2, 0.33, 0.45, 0.52, 0.6, 0.85};
+    std::vector<double> krel = {0.,
+                                0.0022037808641975302,
+                                0.030140817901234556,
+                                0.080908641975308573,
+                                0.19753086419753069,
+                                1.0};
+
+    const double purterbation = 1.e-9;
+    for (std::size_t i = 0; i < S.size(); i++)
+    {
+        ASSERT_NEAR(krel[i], perm_model->getValue(S[i]), 1.e-9);
+
+        // i ==  S.size() -1, S(=0.85) is limited to 0.8, and the numerical
+        // derivative of krel is unavailable.
+        if (i == S.size() - 1)
+            continue;
+        // Compare the derivative with numerical one.
+        ASSERT_NEAR((perm_model->getValue(S[i] + purterbation) -
+                     perm_model->getValue(S[i])) /
+                        purterbation,
+                    perm_model->getdValue(S[i]), 1.e-6);
+    }
+}
+
+TEST(MaterialPorousMedium, checkNonWettingPhaseBrookCoreyOilGas)
+{
+    const char xml[] =
+        "<relative_permeability id=\"0\">"
+        "   <type>NonWettingPhaseBrookCoreyOilGas</type>"
+        "   <sr>  0.2 </sr> "
+        "   <smax> 0.8 </smax> "
+        "   <m> 2 </m> "
+        "   <krel_min> 1.e-9 </krel_min> "
+        "</relative_permeability>";
+    auto const perm_model = createRelativePermeabilityModel(xml);
+
+    std::vector<double> S = {0.2, 0.33, 0.45, 0.52, 0.6, 0.85};
+    std::vector<double> krel = {1.0,
+                                0.58480547839506147,
+                                0.28120177469135793,
+                                0.15583209876543211,
+                                0.061728395061728412,
+                                .0};
+
+    const double purterbation = 1.e-9;
+    for (std::size_t i = 0; i < S.size(); i++)
+    {
+        ASSERT_NEAR(krel[i], perm_model->getValue(S[i]), 1.e-9);
+
+        // Compare the derivative with numerical one.
+        ASSERT_NEAR((perm_model->getValue(S[i] + purterbation) -
+                     perm_model->getValue(S[i])) /
+                        purterbation,
+                    perm_model->getdValue(S[i]), 1.e-6);
+    }
+}
+
+TEST(MaterialPorousMedium, checkReletivePermeabilityCurve)
+{
+    const char xml[] =
+        "<relative_permeability id=\"0\">"
+        "   <type>Curve</type>"
+        "       <curve>"
+        "           <point><data> 0. 0.9   </data></point>"
+        "           <point><data> 0.4 0.5  </data></point>"
+        "           <point><data> 0.9 0.01 </data></point>"
+        "       </curve>"
```

Comment:
See my message to Christoph:
```
 For the curve, when I started to implement the curve type material model,
 I found that the curve should be placed to where it used or belong to. 
 Parsing all curves and saved them as a member of ProjectData
 is not the best choice. Material models can have curves, BCs can 
have curves. 
These curves should be encapsulated to the classes that use them.

Besides, input the variables and values in separated vector is not 
good to perform a manual input.

Therefore, I did not add the parsing of the curve for relative
 permeability to ProjectData class.
```

If you and others think it is not good to have this XML syntax for a curve. I can use the two array type input by extracting  createPiecewiseLinearInterpolation(BaseLib::ConfigTree const& config) from Project Data and saving it to a single file in MathLib/InterpolationAlgorithms.